src/Compositor.cpp

```diff
@@ -552,6 +552,9 @@ void CCompositor::initManagers(eManagersInitStage stage) {
 
             g_pConfigManager->init();
             g_pWatchdog = std::make_unique<CWatchdog>(); // requires config
+            // wait for watchdog to initialize to not hit data races in reading config values.
+            while (!g_pWatchdog->m_bWatchdogInitialized)
+                std::this_thread::yield();
```

Comment:
what? sleep for a ms?

Reply:
Hm took it from here https://cplusplus.com/reference/thread/this_thread/yield/

Reply:
That example section

Reply:
fair enough